reputation/plugin_bot.go

```diff
@@ -63,8 +63,20 @@ func handleMessageCreate(evt *eventsystem.EventData) {
 		return
 	}
 
-	if !isThanksDetectionAllowedInChannel(conf, msg.ChannelID) {
-		return
+	cState := evt.CSOrThread()
+	if cState == nil {
+		return // No channel state, ignore
+	}
+
+	// Check if thanks detection is allowed in the parent channel
+	if cState.Type.IsForum() || cState.Type.IsThread() {
```

Comment:
`cState.Type.IsForum()` returns true if the channel is a forum channel, not a forum post. Forum posts are public threads, so `IsThread` is all you need here I believe. Have you tested this?

Reply:
No, I haven't tested with `cState.Type. IsThread()` only

<img width="459" alt="image" src="https://github.com/botlabs-gg/yagpdb/assets/20161393/76065625-ca3b-4765-b933-8355a5197256">

Let me check

Reply:
Yes, you're right, `IsThread` is all I need